server/store/layer_generators/timer_layer.go.tmpl

```diff
@@ -0,0 +1,52 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+// Code generated by "make generate"
+// DO NOT EDIT
+
+package timerlayer
+
+import (
+        "time"
+	"database/sql"
+
+	"github.com/mattermost/mattermost-plugin-msteams-sync/server/metrics"
+	"github.com/mattermost/mattermost-plugin-msteams-sync/server/store"
+	"github.com/mattermost/mattermost-plugin-msteams-sync/server/store/storemodels"
+
+	"golang.org/x/oauth2"
+)
+
+type {{.Name}} struct {
+	store.Store
+	metrics *metrics.Metrics
+}
+
+{{range $index, $element := .Methods}}
+func (s *{{$.Name}}) {{$index}}({{$element.Params | joinParamsWithType}}) {{$element.Results | joinResultsForSignature}} {
+	start := time.Now()
+	{{if $element.Results | len | eq 0}}
+	s.Store.{{$index}}({{$element.Params | joinParams}})
+	{{else}}
+	{{genResultsVars $element.Results false }} := s.Store.{{$index}}({{$element.Params | joinParams}})
+	{{end}}
+	elapsed := float64(time.Since(start)) / float64(time.Second)
+        success := "false"
+        if {{$element.Results | errorToBoolean}} {
+                success = "true"
+        }
+        s.metrics.ObserveStoreMethodDuration("Store.{{$index}}", success, elapsed)
+	{{ with (genResultsVars $element.Results false ) -}}
+	return {{ . }}
+	{{- else -}}
+	}
```

Comment:
```suggestion
```
Can you verify if there's a need for this `else` condition or not?

Reply:
I believe this handles the case when the store method returns nothing. I can't find a case where that happens today, but I think it makes sense for the layer generator to handle it seamlessly regardless.

Reply:
When I was working on this template file for the client package, I got error here. If a function is not returning anything, then this `else` condition adds an extra `}` that gives error.

Reply:
Oh, I see! My apologies, I didn't follow the thread fully.